pandas/tseries/timedeltas.py

```diff
@@ -142,13 +137,50 @@ def _validate_timedelta_unit(arg):
 
 
 def _coerce_scalar_to_timedelta_type(r, unit='ns', box=True, errors='raise'):
-    """
-    convert strings to timedelta; coerce to Timedelta (if box), else
-    np.timedelta64
-    """
+    """Convert string 'r' to a timedelta object."""
+
+    try:
+        result = tslib.convert_to_timedelta64(r, unit)
+    except Exception:
+        if errors == 'raise':
+            raise
+        elif errors == 'ignore':
+            return r
+
+        # coerce
+        result = pd.NaT
 
-    result = tslib.convert_to_timedelta(r, unit, errors)
```

Comment:
This layout is not correct.  The reason why the `coerce` handling moved to the Python level is because of the `box` check.  In the current framework, if `box=True` and `errors='ignore'` but parsing fails, the conversion to `Timedelta` will raise when it shouldn't.  We could move the `box` check into `convert_to_timedelta(64)`, but that seems to be putting too much into one function, not to mention this is also called on the `array_to_timedelta` end.